### ISSUE TITLE
DOOM 3 (9050) - Working Package

### ISSUE THREAD

**Issue by d10sfan** [dreamer/luxtorpeda#2]
I created a package based on the Jedi Knight package, using the dhewm3 project. Ran it through the docker image you linked and was able to create a dist build, put it into the cache directory and have Luxtorpeda run and install it. I was at least able to prove that the dhewm3 binary runs and got into a single player campaign start. I also added support for 9070 (the expansion pack).

Wasn't sure exactly where to put the package for review, so I created a repo here: https://github.com/d10sfan/luxtorpeda-dhewm3

I didn't touch the LICENSE file, but the rest of it should be feature complete to run Doom through this layer.

For 9070, the engine needs to be run with /dhewm3 +set fs_game d3xp. I attempted to have that in the packages.json as a command_arg, but it didn't look like it worked. Running it directly with that from the steam runtime (~/.steam/steam/ubuntu12_32/steam-runtime/run.sh ./dhewm3 +set fs_game d3xp) worked fine. The manual edits to the packages.json are below:

```
        "9050": {
		"game_name":    "DOOM 3",
		"package":      "dhewm3-9050.tar.xz",
		"command":      "./dhewm3"
	},
	"9070": {
		"game_name":    "DOOM 3 EXP",
		"package":      "dhewm3-9070.tar.xz",
		"command":      "./dhewm3",
		"command_args": "+set fs_game d3xp"
	}
```

For building, I did the following (just for ease of use):

- sudo docker run -i -t luxtorpeda/steam-runtime-amd64-beta:test-1 /bin/bash
- apt-get install git (Just to download the repo so not to worry about the transfer of files back and forth)
- git clone https://github.com/d10sfan/luxtorpeda-dhewm3.git --recursive
- ./build.sh and then ./package.sh
- Once that's done, I do a docker ps to find the container id, then do a docker cp to get the binaries. 
- Then, copy the binaries archive to ~/.cache/luxtorpeda, manually edit packages.json to have the new games.

**Comment by dreamer** [comment]
It's great, thanks :)

* dhewm3 mirror: https://gitlab.com/luxtorpeda/mirrors/dhewm3 - I create mirrors of projects for packaging to speed up GitLab builds and to simplify patch distribution, if it'll ever be required.
* package: https://gitlab.com/luxtorpeda/packages/dhewm3 - I had to do tiny changes for CI. Let me know if you want to be added as package maintainer to this repo.

> I didn't touch the LICENSE file …

This is something I'll need to improve in future iterations - I need to make it clear for end users which license covers package scripts and which license covers the engine itself, distribute licenses texts with the package, etc.

> I attempted to have that in the packages.json as a command_arg, but it didn't look like it worked.

Yeah - it worked in one of initial versions, then I needed to temporarily remove it when implementing `packages.json` file. I just re-implemented it back, the latest version from master seems to be working ok (but I tested with Q3A, as I don't own expansion pack for Doom 3).

> sudo docker run …

I intend to document this process together with instructions for locally rebuilding docker image at https://gitlab.com/luxtorpeda/runtime As for git (and few other tools) - I think I'll include them in some future release of docker image - missing git binaries prevented me from generating useful metadata to be included with the packages (that's why `manifest` file is empty).

BTW, I looked at dhewm3 releases and it seems like next version will support Doom 3 demo: https://steamdb.info/app/9100/

I'll close this issue as soon as I'll get confirmation that new version of Luxtorpeda works with Doom 3 expansion pack :)
